homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -0,0 +1,359 @@
+DEFAULT_RADIUS_IN_KM = 20.0
+DEFAULT_SCAN_INTERVAL = timedelta(minutes=5)
+DEFAULT_UNIT_OF_MEASUREMENT = 'Events'
+
+DOMAIN = 'geo_rss_events'
+ENTITY_ID_FORMAT = 'sensor.' + DOMAIN + '_{}'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Optional(CONF_RADIUS, default=DEFAULT_RADIUS_IN_KM): vol.Coerce(float),
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Optional(CONF_ICON, default=DEFAULT_ICON): cv.icon,
+    vol.Optional(CONF_CATEGORIES, default=[]): vol.All(cv.ensure_list,
+                                                       [cv.string]),
+    vol.Optional(CONF_UNIT_OF_MEASUREMENT,
+                 default=DEFAULT_UNIT_OF_MEASUREMENT): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices,
+                         discovery_info=None):  # pragma: no cover
+    """Set up the GeoRSS component."""
+    # Grab location from config
+    home_latitude = hass.config.latitude
+    home_longitude = hass.config.longitude
+    url = config.get(CONF_URL)
+    radius_in_km = config.get(CONF_RADIUS)
+    name = config.get(CONF_NAME)
+    icon = config.get(CONF_ICON)
+    categories = config.get(CONF_CATEGORIES)
+    scan_interval = config.get(CONF_SCAN_INTERVAL) or DEFAULT_SCAN_INTERVAL
+    unit_of_measurement = config.get(CONF_UNIT_OF_MEASUREMENT)
+
+    if None in (home_latitude, home_longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    _LOGGER.debug("latitude=%s, longitude=%s, url=%s, radius=%s",
+                  home_latitude, home_longitude, url, radius_in_km)
+
+    data = GeoRssServiceData()
+
+    # Create all sensors based on categories.
+    devices = []
+    if not categories:
+        device = GeoRssServiceSensor(hass, None, [], name, icon,
+                                     unit_of_measurement)
+        devices.append(device)
+    else:
+        for category in categories:
+            device = GeoRssServiceSensor(hass, category, data, name, icon,
+                                         unit_of_measurement)
+            devices.append(device)
+    async_add_devices(devices)
+
+    # Initialise update service.
+    updater = GeoRssServiceUpdater(hass, data, home_latitude, home_longitude,
+                                   url, radius_in_km, devices)
+    async_track_time_interval(hass, updater.async_update, scan_interval)
+    yield from updater.async_update()
+    return True
+
+
+class GeoRssServiceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, hass, category, data, name, icon,
+                 unit_of_measurement):
+        """Initialize the sensor."""
+        self.hass = hass
+        self._category = category
+        self._data = data
+        self._state = STATE_UNKNOWN
+        self._name = name
+        self._icon = icon
+        self._unit_of_measurement = unit_of_measurement
+        if category is not None:
```

Comment:
Small style preference. Would prefer one of:
`id_base = category or 'any'`
`id_base = 'any' if category is None else category`